pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -936,16 +945,21 @@ func (c *Controller) updateEDS(ep *v1.Endpoints, event model.Event) {
 			for _, ea := range ss.Addresses {
 				pod := c.pods.getPodByIP(ea.IP)
 				if pod == nil {
-					// This can not happen in usual case
+					// This means, the endpoint event has arrived before pod event. This might happen because
+					// PodCache is eventually consistent. We should try to get the pod from underlying store.
+					endpointBeforePodEvents.Increment()
```

Comment:
It maybe cause by no related pod at all, move down the if branch?

Reply:
I thought it would be useful capture how many times we get endpoints before pod comes. So this metric tracks. The one inside if block captures endpoints without pods. So both may be useful. WDYT?

Reply:
I mean there are cases, no pod at all, just an endpoint. so this cal is not correct

Reply:
ok. I removed that metric